powerscale/provider/cluster_email_resource.go

```diff
@@ -0,0 +1,402 @@
+	"regexp"
+	"terraform-provider-powerscale/client"
+	"terraform-provider-powerscale/powerscale/constants"
+	"terraform-provider-powerscale/powerscale/helper"
+	"terraform-provider-powerscale/powerscale/models"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var (
+	_ resource.Resource                = &ClusterEmailResource{}
+	_ resource.ResourceWithConfigure   = &ClusterEmailResource{}
+	_ resource.ResourceWithImportState = &ClusterEmailResource{}
+)
+
+// NewClusterEmailResource creates a new resource.
+func NewClusterEmailResource() resource.Resource {
+	return &ClusterEmailResource{}
+}
+
+// ClusterEmailResource defines the resource implementation.
+type ClusterEmailResource struct {
+	client *client.Client
+}
+
+// Metadata describes the resource arguments.
+func (r *ClusterEmailResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_cluster_email"
+}
+
+// Schema describes the resource arguments.
+func (r *ClusterEmailResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This resource is used to manage the Cluster Email Settings entity of PowerScale Array. " +
+			"PowerScale Cluster Email Settings provide the ability to configure email settings on the cluster." +
+			"We can Create, Update and Delete the Cluster Email Settings using this resource. We can also import existing Cluster Email Settings from PowerScale array. " +
+			"Note that, Cluster Email Settings is the native functionality of PowerScale. When creating the resource, we actually load Cluster Email Settings from PowerScale to the resource state. ",
+		Description: "This resource is used to manage the Cluster Email Settings entity of PowerScale Array. " +
+			"PowerScale Cluster Email Settings provide the ability to configure email settings on the cluster." +
+			"We can Create, Update and Delete the Cluster Email Settings using this resource. We can also import existing Cluster Email Settings from PowerScale array. " +
+			"Note that, Cluster Email Settings is the native functionality of PowerScale. When creating the resource, we actually load Cluster Email Settings from PowerScale to the resource state. ",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "ID of the Cluster Email Settings.",
+				MarkdownDescription: "ID of the Cluster Email Settings.",
+				Computed:            true,
+			},
+			"settings": schema.SingleNestedAttribute{
+				Description:         "Cluster email notification settings.",
+				MarkdownDescription: "Cluster email notification settings.",
+				Required:            true,
```

Comment:
I remember that for single resource, better use optional not required.

Reply:
Thinking that for the "setting" resource, empty body block makes no sense. Whatever provided in the tf file should be synchronized with the powerscale side. As long as user wants to manage the cluster email resource, this attribute is needed. Besides, all the attributes of cluster email are included in this `settings` nested attribute, so make it required for better usage.